src/views/News/index.js

```diff
@@ -76,6 +76,7 @@ const StudentNews = (props) => {
   const [loading, setLoading] = useState(true);
   const [categories, setCategories] = useState([]);
   const [news, setNews] = useState([]);
+  const [allNews, setAllNews] = useState([]);
```

Comment:
I think a ref is more appropriate for this use case, since allNews gets set once when the component is first created, and then never gets changed. Using a state variable implies that this value will change overtime and React should re-render when it does. A ref, on the other hand, doesn't cause a re-render when it changes and it's value persists throughout all renders. This will clarify that allNews is not dynamic state, but rather is being set on load and only read thereafter.

You can switch this to a ref bu updating this line:
 ```suggestion
  const allNewsRef = useRef([]);
```

Then, when you call setAllNews, instead do `allNewsref.current = unexpiredNews`, and when you reference `allNews`, change it to `allNewsRef.current`.

Reply:
Ah, I was not aware of the distinction. Thank you!